themes/bootstrap3/templates/RecordDriver/LibGuides/result-list.phtml

```diff
@@ -10,5 +10,12 @@
         </a>
       </h2>
     </div>
+    <?php foreach ($this->driver->getSummary() as $summaryItem): ?>
+      <?php if (!empty($summaryItem)): ?>
+        <div class="resultItemLine1">
+          <?=$this->escapeHtml($summaryItem)?>
```

Comment:
Based on the way the summary is getting extracted from HTML, is it possible that it may already include escaped characters or markup? We should be sure there is no possibility of double-escaping anything (or letting in dangerous unsanitized HTML, for that matter).

Reply:
Yeah that's a good point.  But I definitely see unescaped characters in the data coming in from the widget HTML, i.e. the ampersand in `<a href="https://libraryguides.lehigh.edu/electrical_engineering" target="_blank">Electrical & Computer Engineering<i class="fa fa-external-link pad-left-sm"></i></a>`.  So I think it is correct to escape here.

Reply:
Is it possible that you should `html_entity_decode` the text extracted from the widget HTML to be on the safe side? Do you have the ability to create a test description containing HTML or at least `<` / `>` characters and see how it comes out in the widget?

Reply:
I went into the LibGuides admin UI and added HTML to bold a word in a database description.  When pulled into VuFind, the HTML is rendered.

![image](https://github.com/vufind-org/vufind/assets/31278545/c990c95d-43e7-44ea-a17f-72ee23d3ad8b)

I think(?) that is evidence that a) the current escaping is sufficient and b) no one should use HTML in these descriptions in the first place.  But feel free to push back if you still think html_entity_decode is needed somewhere -- I'm not clear on that.

Reply:
How does this description display in LibGuides itself? I think our goal should be to mirror that as much as possible. So if the bold word is bold in LibGuides but shows as HTML in VuFind, we need to fix that. But if it shows as HTML in both places, then I think all is well. I would be interested in how a description like `< &lt; &gt; > & &amp;` shows up in both places too -- curious about handling of HTML entities and unescaped special HTML characters.

Reply:
Yeah, the description does get bolded within LibGuides itself.

![image](https://github.com/vufind-org/vufind/assets/31278545/6b7ffc9c-a06a-4849-87a0-8eae5306fcd9)

But I assumed actually parsing the entities as entities would be a problem, since `<script>` etc., although I guess the CSP policy with nonce would alleviate some of that risk.

Reply:
I'll try `html_entity_decode` and see if that improves things.

Reply:
You could also use `strip_tags` to get rid of HTML tags (in addition to `html_entity_decode` to handle the entities) if you want to avoid the issue of displaying HTML safely. Obviously not a perfect solution, but maybe the easiest compromise.

Reply:
Want to borrow our [cleanHtml](https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/View/Helper/Root/CleanHtml.php) helper? :)

Reply:
Good call and fixed.